tests/test_build.py

```diff
@@ -40,6 +40,21 @@ def test_respects_none_overrides() -> None:
     assert result.hobbies is None
 
 
+def test_allows_nested_optional_field_overrides() -> None:
+    class MyChildModel(BaseModel):
+        name: str
+
+    class MyParentModel(BaseModel):
+        child: MyChildModel | None
```

Comment:
```suggestion
        child: Optional[MyChildModel]
```

Since we support older Python versions, this syntax won't work when we actually end up evaluating the types. This is the reason the tests are failing.

Reply:
https://github.com/litestar-org/polyfactory/pull/420/commits/cf6641ef801de153e65d74d9174edbe65289dc26